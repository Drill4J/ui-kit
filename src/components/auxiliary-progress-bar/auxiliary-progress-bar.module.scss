@import '../../style-guide/index';

.auxiliary-progress-bar {
  height: 12px;
  border-radius: 4px;
  background: map-get($monochrome, light-tint);

  &__progress {
    height: 12px;
    border-radius: 4px;
    background: map-get($data-visualization, build-cover);
    transition: width 2000ms ease-in-out;

    &--type {
      border-radius: 0 4px 4px 0;

      &:hover {
        transform: scaleY(1.7);
      }

      &--primary {
        background: map-get($data-visualization, scope-cover);
      }

      &--secondary {
        background: map-get($data-visualization, overlapping);
      }
    }
  }
}
