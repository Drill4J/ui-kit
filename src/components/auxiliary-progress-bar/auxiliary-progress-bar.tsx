import * as React from 'react';
import { BEM } from '@redneckz/react-bem-helper';

import styles from './auxiliary-progress-bar.module.scss';

interface Props {
  className?: string;
  value: string;
  type?: 'PRIMARY' | 'SECONDARY';
  testContext?: string;
}

const auxiliaryProgressBar = BEM(styles);

export const AuxiliaryProgressBar = auxiliaryProgressBar(({
  className, value, type, testContext,
}: Props) => (
  <div className={className} data-test={`auxiliary-progress-bar:${(testContext || type)}`}>
    <Progress style={{ width: value }} type={type} />
  </div>
));

const Progress = auxiliaryProgressBar.progress('div');
