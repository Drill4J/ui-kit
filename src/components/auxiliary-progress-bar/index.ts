export { AuxiliaryProgressBar } from './auxiliary-progress-bar';
