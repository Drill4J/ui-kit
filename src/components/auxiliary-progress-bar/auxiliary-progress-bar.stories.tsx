import React from 'react';
import { storiesOf } from '@storybook/react';

import { AuxiliaryProgressBar } from './auxiliary-progress-bar';

storiesOf('AuxiliaryProgressBar', module)
  .add('AuxiliaryProgressBar with change coverage panel', () => {
    const [coverage, setCoverage] = React.useState(0);
    const handleChange = ({ target: { value } }: React.ChangeEvent<HTMLInputElement>) => {
      setCoverage(Number(value));
    };
    return (
      <div style={{ display: 'grid', gap: '20px', padding: '40px' }}>
        <AuxiliaryProgressBar value={`${coverage}%`} />
        <AuxiliaryProgressBar value={`${coverage}%`} type="PRIMARY" />
        <AuxiliaryProgressBar value={`${coverage}%`} type="SECONDARY" />
        <input type="range" min="0" max="100" value={coverage} onChange={handleChange} />
      </div>
    );
  });
