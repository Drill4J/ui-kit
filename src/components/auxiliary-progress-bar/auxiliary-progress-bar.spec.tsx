import React from 'react';
import renderer from 'react-test-renderer';

import { AuxiliaryProgressBar } from './auxiliary-progress-bar';

describe('SecondaryProgressBar', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<AuxiliaryProgressBar value="50px" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
