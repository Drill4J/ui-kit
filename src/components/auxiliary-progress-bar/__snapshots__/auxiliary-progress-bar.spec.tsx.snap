// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryProgressBar should match snapshot 1`] = `
<div
  className="auxiliary-progress-bar"
  data-test="auxiliary-progress-bar:undefined"
>
  <div
    className="auxiliary-progress-bar__progress"
    data-modifiers=""
    style={
      Object {
        "width": "50px",
      }
    }
  />
</div>
`;
